src/inmanta/data/__init__.py

```diff
@@ -2937,6 +2937,75 @@ async def get_deleted_resources(cls, environment, current_version, current_resou
 
         return should_purge
 
+    @classmethod
+    async def get_resource_details(cls, env: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional[m.ResourceDetails]:
+        status_subquery = """
+        (CASE WHEN
+            (SELECT resource.model < MAX(configurationmodel.version)
```

Comment:
can't this simply be latest.version?

Reply:
`latest.version` is only the latest with regards to the current resource (the one the query is about). 
To decide if it's orphaned we need the 'absolute' latest released version (within the environment). 

(It's true that for the status of the required resources, we could use that if they have the same version as the main resource, they must be orphaned as well, but I think Postgres might optimize this in any case).